pyproject.toml

```diff
@@ -55,7 +55,260 @@ pytest-cov = "^3.0.0"
+_format-firmware-font = "black ./firmware/font"
+_format-firmware-scripts = "black ./firmware/scripts"
+_format-firmware = ["_format-firmware-font", "_format-firmware-scripts"]
+_format-i18n = "black ./i18n/i18n.py"
+_format-simulator = "black ./simulator/simulator.py"
+_format-src = "black ./src"
+_format-tests = "black ./tests"
+format = ["_format-firmware", "_format-i18n", "_format-simulator", "_format-src", "_format-tests"]
+
+# tasks for lint code with pylint
+# all tasks that start with `_` are'nt callable
+_lint-firmware-font = "pylint ./firmware/font/*.py"
+_lint-firmware-scripts = "pylint ./firmware/scripts/*.py"
+lint-firmware = ["_lint-firmware-font", "_lint-firmware-scripts"]
+lint-i18n = "pylint ./i18n/i18n.py"
+_lint-simulator-kruxsim = "pylint ./simulator/kruxsim"
+_lint-simulator-src = "pylint ./simulator/simulator.py"
+lint-simulator = ["_lint-simulator-kruxsim", "_lint-simulator-src"]
+lint-src = "pylint ./src"
+lint-tests = "pylint ./tests"
+
+# tasks for lint markdown texts code with pymarkdownlnt
+# all tasks that start with `_` are'nt callable
+_lint-md-readme = "pymarkdown scan README.md"
+_lint-md-changelog = "pymarkdown scan CHANGELOG.md"
+_lint-md-docs = "pymarkdown scan docs"
+lint-markdown = ["_lint-md-readme","_lint-md-changelog", "_lint-md-docs"]
+
+# Lint all
+lint = ["lint-firmware", "lint-i18n", "lint-simulator", "lint-src", "lint-tests", "lint-markdown"]
+
+# tasks for test code
+test = "poetry run pytest --cache-clear --cov src/krux --cov-report html ./tests"
+test-verbose = "poetry run pytest --cache-clear --cov src/krux --cov-report html --show-capture all --capture tee-sys -r A ./tests"
+test-specific = "poetry run pytest --cache-clear" # Need extra arguments
+
+# tasks for simulation
+# put --sd if you want to simulate sd
+simulate-m5stickv = "python simulator/simulator.py --device maixpy_m5stickv" 
+simulate-amigo-tft = "python simulator/simulator.py --device maixpy_amigo_tft"
+simulate-amigo-ips = "python simulator/simulator.py --device maixpy_amigo_ips"
+simulate-dock = "python simulator/simulator.py --device maixpy_dock"
+simulate-pc = "python simulator/simulator.py --device maixpy_pc"
+
+
+# tasks for live debug
+live-debug = "screen /dev/tty.usbserial-device-name 115200"
+
+# tasks for build docs
+docs-serve = "mkdocs serve"
+docs-build = "mkdocs build --output" # Need an extra argument
+
+# tasks for generate screenshots
+_rmdir-screenshot = "rm -rf simulator/screenshots"
+_mkdir-screenshot = "mkdir -p simulator/screenshots"
+_rmdir-sd = "rm -rf simulator/sd"
+_mkdir-sd = "mkdir -p simulator/sd"
+_make-cfg = "echo \"{\\\"settings\\\": {\\\"i18n\\\": {\\\"locale\\\": \\\"en_US\\\"}}}\" > simulator/sd/settings.json"
+simulate-prepare = [
+  "_rmdir-screenshot",
+  "_mkdir-screenshot",
+  "_rmdir-sd",
+  "_mkdir-sd",
+  "_make-cfg",
+]
+
+# m5stickv
+simulate-sequence-m5stickv-about = "python simulator/simulator.py --sequence simulator/sequences/about.txt  --device maixpy_m5stickv"
```

Comment:
We could use more the variables and less fixed names, so we have more code reuse instead of copy/paste. For example, when you use a simulator sequence you use the start of the name of the file and hit tab to use the terminal autocomplete feature, that way you don't need to figure the exact name of the file. I don't know if the terminal will autocomplete this name `simulate-sequence-m5stickv-bitcoin-options` but we could remember about `simulate-sequence` and then only pass the parameters (the sequence file and the device). Or `simulate-sequence-m5stickv` and only pass the sequence file

Reply:
I can fix this changing the `.sh` files, but i was afraid to break the procedures. So i coded in this manner to see what would you say.